contracts/standard/arbitration/composed-arbitrable/fee/MultiPartyInsurableFees.sol

```diff
@@ -0,0 +1,270 @@
+                        "It is the winning side's turn to fund the appeal, only if the losing side already fully funded it."
+                    );
+            } else require(msg.value >= fundDisputeCache.cost, "Fees must be paid in full if the arbitrator does not support `appealPeriod`.");
+        }
+
+        // Compute required value.
+        if (_side == 0) // Losing side.
+            fundDisputeCache.requiredValueForSide = !fundDisputeCache.appealing ? fundDisputeCache.cost / 2 : fundDisputeCache.cost + (2 * _paidFees.stake[_paidFees.stake.length - 1]);
+        else { // Winning side.
+            fundDisputeCache.expectedValue = _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][0] - _paidFees.stake[_paidFees.stake.length - 1];
+            fundDisputeCache.requiredValueForSide = fundDisputeCache.cost > _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][0] + fundDisputeCache.expectedValue ? fundDisputeCache.cost - _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][0] : fundDisputeCache.expectedValue;
+        }
+
+        // Take contribution.
+        if (_paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][_side] >= fundDisputeCache.requiredValueForSide)
+            fundDisputeCache.stillRequiredValueForSide = 0;
+        else 
+            fundDisputeCache.stillRequiredValueForSide = fundDisputeCache.requiredValueForSide - _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][_side];
+        fundDisputeCache.keptValue = fundDisputeCache.stillRequiredValueForSide >= msg.value ?
+            msg.value : fundDisputeCache.stillRequiredValueForSide;
+        fundDisputeCache.refundedValue = msg.value - fundDisputeCache.keptValue;
+        if (fundDisputeCache.keptValue > 0) {
+            _paidFees.totalValue[_paidFees.totalValue.length - 1] += fundDisputeCache.keptValue;
+            _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][_side] += fundDisputeCache.keptValue;
+            _paidFees.contributions[_paidFees.contributions.length - 1][msg.sender][_side] += fundDisputeCache.keptValue;
+        }
+        if (fundDisputeCache.refundedValue > 0) msg.sender.transfer(fundDisputeCache.refundedValue);
+        emit Contribution(_agreementID, _paidFees.stake.length - 1, msg.sender, fundDisputeCache.keptValue);
+
+        // Check if enough funds have been gathered and act accordingly.
+        if (
+            _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][_side] >= fundDisputeCache.requiredValueForSide ||
+            (fundDisputeCache.appealing && !fundDisputeCache.appealPeriodSupported)
+        ) {
+            if (_side == 0 && !(fundDisputeCache.appealing && !fundDisputeCache.appealPeriodSupported)) { // Losing side and not direct appeal.
+                if (!_paidFees.loserFullyFunded[_paidFees.loserFullyFunded.length - 1])
```

Comment:
Why not: `_paidFees.loserFullyFunded[_paidFees.loserFullyFunded.length - 1] = true;`

instead of: 
```
if (!_paidFees.loserFullyFunded[_paidFees.loserFullyFunded.length - 1])
_paidFees.loserFullyFunded[_paidFees.loserFullyFunded.length - 1] = true;
```

Reply:
Fixed in `a78b94f`.

Reply:
`sstore` is 100 gas. `sload` is only 20.

Reply:
Hmm.

Do you think that the following is equivalent gas-wise?
```
_paidFees.loserFullyFunded[_paidFees.loserFullyFunded.length - 1] = _paidFees.loserFullyFunded[_paidFees.loserFullyFunded.length - 1] || true;
```